Library/Homebrew/formula.rb

```diff
@@ -540,23 +540,13 @@ def oldname
   # Old names for the formula.
   sig { returns(T::Array[String]) }
   def oldnames
-    @oldnames ||= if tap
-      T.must(tap).formula_oldnames.fetch(name, [])
-    else
-      []
-    end
+    @oldnames ||= tap&.formula_oldnames&.dig(name) || []
```

Comment:
Do we expect `tap` to ever be not `nil` but `formula_oldnames` is nil`?

If not, the existing code but just without the `T.must` feels nicer to me than repeatedly chaining `&`